Source/Fuse.Controls.WebView/Android/FuseWebViewClient.java

```diff
@@ -16,30 +17,27 @@
 	Action _urlChangedAction; 
 	Action_String _onCustomURI;
 	String[] _customURIs;
+	Func _hasUriSchemeHandler;
 	
-	public FuseWebViewClient(Action loaded, Action started, Action changed, Action_String onCustomURI, StringArray customURIs)
+	public FuseWebViewClient(Action loaded, Action started, Action changed, Action_String onCustomURI, StringArray customURIs, Func hasUriSchemeHandler)
 	{
 		super();
 		_pageLoadedAction = loaded;
 		_pageStartedAction = started;
 		_urlChangedAction = changed;
 		_onCustomURI = onCustomURI;
 		_customURIs = customURIs.copyArray();
+		_hasUriSchemeHandler = hasUriSchemeHandler;
 		loadingFinished = true;
 		redirect = false;
 	}
 	
 	@Override
 	public boolean shouldOverrideUrlLoading(WebView view, String url) 
 	{
-		for(String uri : _customURIs)
-		{
-			if(url.indexOf(uri) != -1)
-			{
-				_onCustomURI.run(url);
-				return true;
-			}
-		}
+		if(tryInterceptUriScheme(url))
```

Comment:
Might it not be cleaner to make the function `matchedInterceptUriScheme` and do the call to `_onCustomURI.run` here?